runtime/src/main/java/org/corfudb/util/Utils.java

```diff
@@ -228,10 +204,32 @@ public static long getLogTail(Layout layout, CorfuRuntime runtime) {
         } else if (segment.getReplicationMode() == Layout.ReplicationMode.QUORUM_REPLICATION) {
             throw new UnsupportedOperationException();
         }
-
         return globalLogTail;
     }
 
+    public static long getLogSize(Layout layout, CorfuRuntime runtime, long startAddress, long endAddress) {
+        long size = 0;
+        long segmentSize = 0;
+        long currentEnd = 0;
```

Comment:
I think the converse of this logic would be cleaner (i.e. inver the logic). 

If (segment.start >= startAddress && segment.end <= endAddress) {
//compute
}